design-proposals/nic-hotplug/nic-hotplug.md

```diff
@@ -349,19 +261,27 @@ type VirtualMachineInstanceNetworkInterface struct {
 type InterfaceHotplugStatus struct {
     // Phase are specific phases for the hotplug volume.
     Phase InterfaceHotplugPhase `json:"phase,omitempty"`
-
-    // DetailedMessage has more information in case of failed operations.
-    DetailedMessage string `json:"DetailedMessage,omitempty"`
 }
 
 type InterfaceHotplugPhase string
 
 const InterfaceHotplugPhasePending    InterfaceHotplugPhase = "Pending"    # network configuration did not start yet
 const InterfaceHotplugPhaseInfraReady InterfaceHotplugPhase = "InfraReady" # network configuration phase1 completed (networking infra created and configured)
 const InterfaceHotplugPhaseReady      InterfaceHotplugPhase = "Ready"      # network configuration phase1 and phase2 completed (all is OK from the networking perspective)
-const InterfaceHotplugPhaseFailed     InterfaceHotplugPhase = "Failed"     # plugging the new interface failed. More details on `DetailedMessage`.
 ```
 
+The proposed `VirtualMachineInstanceNetworkInterface` status change is really
+only helpful for IPAM scenarios; the `virt-handler` component would block until
+it realizes the multus dynamic networks controller has already finished
+configuring the pod interface accordingly - since there would otherwise be a
+race between the CNI IPAM plugin and `virt-handler` (virt-handler could see the
+pod interface created but **missing** the IP addresses if its reconcile cycle
+kicked in while CNI IPAM was still configuring the link).
+
+For this reason, we have decided to propose an initial stage, where only L2
```

Comment:
Sounds good to me. But then I do not understand why we need `HotplugStatus` at all with all its phases.

Reply:
Per my understanding, without L3, we do not need this at all. Am I wrong here?
And if we do not need this with L3, lets drop it until we will really need it.

I also like it less, because you use a key to express hotplugs specifically. I would prefer to see something generic which will make sense with or without hotplugs. Or in other words, I would prefer not to care it is a hotplug thing at all.

Reply:
> Per my understanding, without L3, we do not need this at all. Am I wrong here? And if we do not need this with L3, lets drop it until we will really need it.

It really depends on the CNI implementation. If they create the interface w/ whatever defaults, then mutate the interface's attributes, we would still be in this situation, as @AlonaKaplan raised.

I would still rather have the simplest possible approach in first - which is a simple reconcile loop.

> 
> I also like it less, because you use a key to express hotplugs specifically. I would prefer to see something generic which will make sense with or without hotplugs. Or in other words, I would prefer not to care it is a hotplug thing at all.

I'll try to rename this, and find a way to expose it via the API in a reusable way instead.

Good feedback, thanks @EdDev .